padl/transforms.py

```diff
@@ -1372,6 +1459,11 @@ def __getitem__(self, item: Union[int, slice, str]) -> Transform:
     def __len__(self):
         return len(self.transforms)
 
+    def _pd_call_transform(self, arg):
```

Comment:
Why does `Map` need to have its own `_pd_call_transform`? I think we would ideally only define this function once for the base `Transform` class.

Reply:
This `_pd_call_transform` belongs to `Pipeline`. It is because `Pipeline` objects already trace the errors on its `__call__`

Reply:
I see. Is it possible to keep the `try` and `except` only in the `_pd_call_transform` and out of the `__call__` methods? This would reduce the number of places that we would need the `try` and `except` boiler plate.

Reply:
I think not, it is needed to do it inside the `__call__` of the `Pipeline`s because I need to trace the item of the `Pipeline` that is getting the Exception.

Reply:
Can we resolve this?

Reply:
Yes I think so. However, in general I feel like the number of `try` and `except` clauses needed makes the code harder to read. I was hoping there would have been a way to reduce the number of `try`'s and `except`'s.